swatch-metrics/src/main/java/com/redhat/swatch/metrics/service/PrometheusMeteringController.java

```diff
@@ -239,6 +239,35 @@ private void createEventFromDataAndSend(
       role = null;
     }
 
+    boolean is3rdPartyMigrated = Boolean.parseBoolean(labels.get("conversions_success"));
+
+    var matchingTags =
+        SubscriptionDefinition.getAllProductTagsWithPaygEligibleByRoleOrEngIds(
+            role, productIds, null, is3rdPartyMigrated);
+
+    Optional<String> firstTag = matchingTags.stream().findFirst();
+    if (firstTag.isPresent()) {
+
+      var derivedProductTag = firstTag.get();
+
+      if (!Objects.equals(derivedProductTag, productTag)) {
+        log.warn(
+            "Starting product tag {} does not match derived product tag {} based on data contents",
+            productTag,
+            derivedProductTag);
+      }
+
+      productTag = derivedProductTag;
```

Comment:
I don't think we need to do this assignment since the producttag and derivedProductTag will be same

Reply:
Changed structure of code so derivedProductTag is no longer scoped to this if statement and can be used further down